bladegps.c

```diff
@@ -428,33 +425,36 @@ int main(int argc, char *argv[])
 		printf("TX sample rate: %u sps\n", TX_SAMPLERATE);
 	}
 
-	s.status = bladerf_set_bandwidth(s.tx.dev, BLADERF_MODULE_TX, TX_BANDWIDTH, NULL);
+	s.status = bladerf_set_bandwidth(s.tx.dev, tx_channel, TX_BANDWIDTH, NULL);
 	if (s.status != 0) {
 		fprintf(stderr, "Failed to set TX bandwidth: %s\n", bladerf_strerror(s.status));
 		goto out;
 	}
 	else {
 		printf("TX bandwidth: %u Hz\n", TX_BANDWIDTH);
 	}
-
-	//s.status = bladerf_set_txvga1(s.tx.dev, TX_VGA1);
-	s.status = bladerf_set_txvga1(s.tx.dev, txvga1);
-	if (s.status != 0) {
-		fprintf(stderr, "Failed to set TX VGA1 gain: %s\n", bladerf_strerror(s.status));
+	
+    	// What is the valid range?
+    	s.status = bladerf_get_gain_range(s.tx.dev, tx_channel, &range);
+    	if (s.status != 0) {
+		fprintf(stderr, "Failed to check gain range: %s\n", bladerf_strerror(s.status));
 		goto out;
-	}
-	else {
-		//printf("TX VGA1 gain: %d dB\n", TX_VGA1);
-		printf("TX VGA1 gain: %d dB\n", txvga1);
-	}
-
-	s.status = bladerf_set_txvga2(s.tx.dev, TX_VGA2);
+    	}
+    	else {
+    		printf("Range: [%g, %g]\n",range->min * range->scale, range->max * range->scale);
```

Comment:
probably worthwhile to create a local variable for min and max since it's used at least twice in this loop.